gpflow/kernels/multioutput/kernels.py

```diff
@@ -73,7 +73,9 @@ def K_diag(self, X, full_output_cov=True):
         """
         raise NotImplementedError
 
-    def __call__(self, X, X2=None, *, full_cov=False, full_output_cov=True):
+    def __call__(self, X, X2=None, *, full_cov=False, full_output_cov=True, presliced=False):
+        if not presliced:
+            X, X2 = self.slice(X, X2)
         if not full_cov and X2 is not None:
             raise ValueError("Ambiguous inputs: `diagonal` and `y` are not compatible.")
```

Comment:
Would you mind updating the error message while we're looking at this?
```suggestion
            raise ValueError("Ambiguous inputs: passing in `X2` is not compatible with `full_cov=False`.")
```